### PR TITLE
Added Hydra support as an option in ArgParse UI 

### LINKED ISSUE #102: [FR] Add google-fire as cmd line config manager as an option
I like google-fire and think it could simplify more the templates. If we can provide it as an option, would be great.

### PR DESCRIPTION
**Pull request by guptaaryan16**
### Hydra support in Code Generator
Fix #102 

This PR is a continuation of python fire PR (#279) and is raised as a draft for testing now. To know more about Hydra as a tool, visit https://hydra.cc

### Additional context

---

### What is the purpose of this pull request?

- [ ] Bug fix
- [X] New feature
- [ ] Other

### CODE REVIEW

File: src/components/TabTraining.vue [hunk 1]
```diff
@@ -1,6 +1,13 @@
 <template>
   <div class="tab training">
-    <h1>Training Options</h1>
+    <h1>Configuraion Options</h1>
```
Comment (vfdev-5) [inline src/components/TabTraining.vue:3]: This header should Training Options, please revert

File: src/templates/template-common/main.py [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
+#::: } else if ((it.argparser == 'hydra')) { :::#
+@hydra.main(version_base=None, config_path=".", config_name="config")
+def main(cfg: DictConfig):
+    config = setup_config(cfg)
+
+
+#::: } else { :::#
 def main():
     config = setup_config()
+    #::: } :::#
```
Comment (vfdev-5) [inline src/templates/template-common/main.py:18]:
```suggestion
#::: } :::#
```

File: src/templates/template-common/utils.py [hunk 3]
```diff
@@ -36,6 +44,50 @@
...
+        if config.get(attr, None) == None:
+            OmegaConf.update(config, attr, None, force_add=True)
+    return config
+
+
+#::: } :::#
+
+
+#::: if ((it.argparser == 'fire')) { :::#
+from omegaconf import DictConfig
```
Comment (vfdev-5) [inline src/templates/template-common/utils.py:65]: This line can be refactored

File: src/templates/template-common/main.py [hunk 4]
```diff
@@ -1,19 +1,34 @@
 # main entrypoint
+#::: if ((it.argparser == 'fire')) { :::#
+def main(config_path, backend=None, **kwargs):
+    config_path = Path(config_path)
+    assert config_path.exists(), config_path
+    config = setup_config(config_path, backend, **kwargs)
+
+
+#::: } else if ((it.argparser == 'hydra')) { :::#
+@hydra.main(version_base=None, config_path=".", config_name="config")
```
Comment (vfdev-5) [inline src/templates/template-common/main.py:10]: I do not understand how to pass backend to the config ?
Which backend is user by default ?

**Review by vfdev-5 — COMMENTED** [review]
I tried to run vision classification template in colab and it fails:
image

**Review by vfdev-5 — COMMENTED** [review]
Removed the comment as it is related to https://github.com/pytorch-ignite/code-generator/pull/279

File: src/templates/template-common/main.py (hunk 4 above)
Comment (guptaaryan16) [reply src/templates/template-common/main.py:10]: By default `backend = None` but if we want to send a specific backend then `python main.py ++backend='gloo'`

**Pull request closed without merging**